ggshield/core/lines.py

```diff
@@ -26,13 +26,40 @@ class LineCategory(Enum):
 
 class Line(NamedTuple):
     """
-    Line object making easier to handle line
-    by line display.
+    Represent a line in a document.
 
     - content: Content of the line
     - category: The line category [+|-| ] (addition, deletion, untouched)
     - pre_index: Line index (deletion for patches, line index for files)
     - post_index: Line index (addition for patches)
+
+    pre_index and post_index are 1-based.
+
+    For files: pre_index is the line number, post_index is None.
+
+    For patches: pre_index and post_index are line numbers relative to the *document
+    modified by the patch*, not to the patch itself. This means that for a patch like
+    this:
+
+    ```
+    diff --git a/foo.py b/foo.py
+    index c53e550a..7ed9d026 100644
+    --- a/foo.py
+    +++ b/foo.py
+    @@ -56,8 +56,8 @@
+     Hello
+    -world
+    +beautiful
+    +WORLD
+    ```
+
+    The values of pre_index and post_index are:
+
+    line          pre_index  post_index
+    " Hello"      56         56
+    "-world"      57         None
+    "+beautiful"  None       57
+    "+WORLD"      None       58
```

Comment:
Actually here in [the lines content the first character is always removed when it is as patch](https://github.com/GitGuardian/ggshield/blob/f9c849ad90b45f8fbb96a241543faac17b4ff8ce/ggshield/core/lines.py#L106-L106)